addons/account/models/partner.py

```diff
@@ -514,6 +514,17 @@ def create(self, vals_list):
                     vals['supplier_rank'] = 1
         return super().create(vals_list)
 
+    def unlink(self):
+        """
+        Prevent the deletion of a partner "Individual", child of a company if:
+        - partner in 'account.move'
+        - state: all states (draft and posted)
+        """
+        moves = self.sudo().with_context(active_test=False).env['account.move'].search_count([('partner_id', '=', self.id), ('state', 'in', ['draft', 'posted'])])
```

Comment:
```suggestion
        moves = self.sudo().with_context(active_test=False).env['account.move'].search_count([('partner_id', 'in', self.ids), ('state', 'in', ['draft', 'posted'])])
```

Reply:
Hello

- For the batch, I'll do it!
- For the `active_test=True` it was just an extra security but, by default, it is indeed not possible
- > But I'm not sure to see which constraint you are talking about.
Could you explain?
-> The `partner_id` has a constraint `ondelete='restrict'` https://github.com/odoo/odoo/blob/feb4e839dc4e35ab16527e5e264b5cc1c0a76c8e/addons/account/models/account_move.py#L2728

Since there is an onchange method (see before) that changes the `partner_id` by the `partner_id.commercial_partner_id` (and the `partner_id.commercial_partner_id` of child is the partner_id of the company associated), there is a need to add this override method.

Reply:
What about this last comment @yosa-odoo ?

Reply:
My bad, I thought that the `ondelete='restrict'` was also not possible in stable. After a double-check, it can be applied in a limited way. I guess we are in the right situation to apply it.

> Limited, compatible changes such as changing ondelete rules or size parameters are allowed when necessary.